openwisp_utils/tasks.py

```diff
@@ -0,0 +1,8 @@
+from celery import Task
+
+from . import settings as app_settings
+
+
+class OpenwispCeleryTask(Task):
+    soft_time_limit = app_settings.CELERY_SOFT_TIME_LIMIT
+    time_limit = app_settings.CELERY_HARD_TIME_LIMIT
```

Comment:
I wrote following tasks to test this 

```python
@shared_task(base=OpenwispCeleryTask)
def exceed_soft_time_limit():
    time.sleep(openwisp_utils_settings.CELERY_SOFT_TIME_LIMIT + 5)
```

And called it in following task

```python
def test_time_limits(self):
        tasks.exceed_soft_time_limit.delay()
        self.assertRaises(SoftTimeLimitExceeded, tasks.exceed_soft_time_limit.delay)
```

But the tests was not working even with eager mode.